src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy

```diff
@@ -113,6 +113,30 @@ class Liberty implements Plugin<Project> {
             dependsOn 'installLiberty'
 
             project.afterEvaluate{
+                // Defining files set in build.gradle as inputs
+                if (!project.liberty.server.configFile.toString().equals('default')) {
+                    inputs.file { project.liberty.server.configFile }
+                } else if (new File("${projectDir}/src/main/liberty/config/server.xml").exists()) {
+                    inputs.file { new File("${projectDir}/src/main/liberty/config/server.xml") }
+                }
+                if (!project.liberty.server.bootstrapPropertiesFile.toString().equals('default')) {
+                    inputs.file { project.liberty.server.bootstrapPropertiesFile }
+                } else if (new File("${projectDir}/src/main/liberty/config/bootstrap.properties").exists()) {
+                    inputs.file { new File("${projectDir}/src/main/liberty/config/bootstrap.properties") }
+                }
+                if (!project.liberty.server.jvmOptionsFile.toString().equals('default')) {
+                    inputs.file { project.liberty.server.jvmOptionsFile }
+                } else if (new File("${projectDir}/src/main/liberty/config/jvm.options").exists()) {
+                    inputs.file { new File("${projectDir}/src/main/liberty/config/jvm.options") }
+                }
+                if (!project.liberty.server.serverEnv.toString().equals('default')) {
+                    inputs.file { project.liberty.server.serverEnv }
+                } else if (new File("${projectDir}/src/main/liberty/config/server.env").exists()) {
+                    inputs.file { new File("${projectDir}/src/main/liberty/config/server.env") }
+                }
+                if (project.liberty.server.configDirectory.exists()) {
```

Comment:
make sure you try this on a project where configDirectory is not set.  i.e., will this NPE?